pkg/util/user/range.go

```
@@ -0,0 +1,164 @@
+package user
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Range errors
+var (
+	ErrParseRange   = errors.New("error parsing range; a range must be of one of the following formats: [n], [-n], [n-], [n:m] where n and m are positive numbers and m is greater than or equal to n")
+	ErrInvalidRange = errors.New("invalid range; a range must consist of positive integers and the upper bound must be greater than or equal to the lower bound")
+)
+
+// Range represents a range of user ids. It can be unbound at either end with a nil value
+// I both From and To are present, To must be greater than or equal to From. Bounds are inclusive
+type Range struct {
+	from *int
+	to   *int
+}
+
+// NewRange creates a new range with lower and upper bound
+func NewRange(from int, to int) (*Range, error) {
+	return (&rangeBuilder{}).From(from, nil).To(to, nil).Range()
+}
+
+// NewRangeTo creates a new range with only the upper bound
+func NewRangeTo(to int) (*Range, error) {
+	return (&rangeBuilder{}).To(to, nil).Range()
+}
+
+// NewRangeFrom creates a new range with only the lower bound
+func NewRangeFrom(from int) (*Range, error) {
+	return (&rangeBuilder{}).From(from, nil).Range()
+}
+
+func parseInt(str string) (int, error) {
+	num, err := strconv.Atoi(str)
+	if err != nil {
+		return 0, ErrParseRange
```

You're kinda swallowing the error here, which might be useful for an end user.